### PR TITLE
Use k8s ?timeoutSeconds param instead of excon timeout on resource watch

### PR DESCRIPTION
**Pull request by jakolehm**

### CODE REVIEW

File: lib/k8s/resource_client.rb [hunk 1]
```diff
@@ -224,20 +224,20 @@ def watch(labelSelector: nil, fieldSelector: nil, resourceVersion: nil, timeout:
...
         method: method,
         path: path,
-        response_class: K8s::API::MetaV1::List,
         query: make_query(
           'labelSelector' => selector_query(labelSelector),
           'fieldSelector' => selector_query(fieldSelector),
           'resourceVersion' => resourceVersion,
-          'watch' => '1'
+          'watch' => '1',
+          'timeoutSeconds' => timeout
```
Comment (kke) [inline lib/k8s/resource_client.rb]: I suppose kube api is fine with nil here. `timeout.to_i` would make nil `0`.

File: lib/k8s/resource_client.rb [hunk 2]
```diff
@@ -224,20 +224,20 @@ def watch(labelSelector: nil, fieldSelector: nil, resourceVersion: nil, timeout:
...
         ),
         response_block: lambda do |chunk, _, _|
           data = JSON.parse(chunk)
           event = K8s::API::MetaV1::WatchEvent.new(data)
           resourceVersion = event.resource&.metadata&.resourceVersion
           yield event
         end,
-        read_timeout: timeout
+        timeout: nil
```
Comment (kke) [inline lib/k8s/resource_client.rb]: I think this may be an invalid (=ignored) option.

Comment (kke) [reply lib/k8s/resource_client.rb]: (should probably be read_timeout or just nothing)

Comment (jakolehm) [reply lib/k8s/resource_client.rb]: `nil` makes excon not to timeout ever. If we don't pass `nil` then default read-timeout value is used.

File: lib/k8s/resource_client.rb (hunk 1 above)
Comment (jakolehm) [reply lib/k8s/resource_client.rb]: If `timeoutSeconds` is not given then Kubernetes api uses the default timeout (usually 30mins).

File: lib/k8s/resource_client.rb (hunk 2 above)
Comment (kke) [reply lib/k8s/resource_client.rb]: I can't find :timeout from excon sources, so this should probably be `read_timeout`

Comment (jakolehm) [reply lib/k8s/resource_client.rb]: Oh yes, good catch.

**Pull request merged**